### ISSUE TITLE
[Request]: enable, or allow to opt-in `rkyv`'s `strict` features

### ISSUE THREAD

**Issue by kwonoj** [wasmerio/wasmer#3695]
### Motivation

https://github.com/swc-project/swc/pull/6922

There are couples of obscure report from plugin author / users of swc that plugin execution fails unexpectedly. It should have various different root cause, but one of them was identified as having different `rustc` compiler version can cause those. Per orignal PR's description and its comment https://github.com/swc-project/swc/pull/6922#issuecomment-1424517156, enabling `strict` in rkyv's serialization / deserialization would possibly resolve some of those issues via rkyv's gaurantee. 

SWC itself tried to apply those changes, however found out underlying `wasmer` (specifically `wasmer-types` in wasmer@2) cannot be built when upstream SWC enables those features.

### Proposed solution

Basically request is enabling, or feature flag to support opt-in for rkyv's strict mode. Since SWC relies on wasmer@2 still, it'd be great if changes can be backported to @2 versions. I may try to make necessary changes myself if it is acceptable (and possible to release a new 2.x patch release).

### Alternatives

SWC could try to fork wasmer@2, but that'll be too heavyweight option.

**Comment by theduke** [comment]
#3693 has enabled strict mode by default, and also introduced a `deserialize_checked()` function that will validate the artifact.

Our current policy is that `2.x` only receives critical security updates, so we are very unlikely to accept a backport and push out a new release.

I recommend upgrading to `3.x`. 
Version `3.2` , which will include the changes, will be released soon.
